include/hipSYCL/sycl/libkernel/sscp/builtins/detail/utils.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of AdaptiveCpp, an implementation of SYCL and C++ standard
+ * parallelism for CPUs and GPUs.
+ *
+ * Copyright The AdaptiveCpp Contributors
+ *
+ * AdaptiveCpp is released under the BSD 2-Clause "Simplified" License.
+ * See file LICENSE in the project root for full license details.
+ */
+// SPDX-License-Identifier: BSD-2-
+#ifndef HIPSYCL_SSCP_UTILS_BUILTINS_HPP
+#define HIPSYCL_SSCP_UTILS_BUILTINS_HPP
+
+#include "../builtin_config.hpp"
+
+#define ACPP_SHMEM_ATTRIBUTE                                                                       \
+  static __attribute__((loader_uninitialized)) __attribute__((address_space(3)))
+
+namespace hipsycl::libkernel::sscp {
+
+template <class Tout, class Tin> Tout bit_cast(Tin x) {
+  Tout result;
+  result = __builtin_bit_cast(Tout, x);
+  return result;
```

Comment:
What is the point of this function? Doesn't seem to add anything of value over just calling `__builtin_bit_cast`

Reply:
I would prefer to keep this in a separate function for now, so that I can explicitly set the cast type.

Reply:
You also do this with `__builtin_bit_cast` using the first argument, no?

Can we at least remove all of the unnecessary lines and just implement it as `return __builtin_bit_cast(Tout, x)` which is exactly the same thing?

Reply:
Ahh true, sorry. Removed the `bit_cast` function from `utils.hpp` and switched to using `__builtin_bit_cast`